src/components/Messages/Messages.jsx

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, useState } from "react";
+import "./stylesMessages";
+import ChatListitem from "./ChatListItem";
+import ChatIntro from "./ChatIntro";
+import ChatWindow from "./ChatWindow";
+import NewChat from "./NewChat";
+
+import {
+  MdDonutLarge,
+  MdChat,
+  MdMoreVert,
+  MdOutlineSearch,
+} from "react-icons/md";
+import MessagesAPI from "../../actions/Api";
+
+import {
+  AppWindow,
+  Avatar,
+  ChatList,
+  ContentArea,
+  Search,
+  Sidebar,
+  SidebarHeader,
+  SidebarHeaderButtons,
+  TextInput,
+} from "./stylesMessages";
+import Api from "../../actions/Api";
+
+const Messages = (props) => {
+  const [chatlist, setChatlist] = useState([]);
+  const [user, setUser] = useState(null);
+  const [activeChat, setActiveChat] = useState({});
+  const [showNewChat, setshowNewChat] = useState(false);
+
+  useEffect(() => {
+    if (user !== null) {
+      return MessagesAPI.onChatList(user.id, setChatlist);
+    }
+  }, [user]);
+
+  const handleNewChat = () => {
+    setshowNewChat(true);
+  };
+
+  const handleLoginData = async (u) => {
+    console.log(u);
+    let newUser = {
+      id: u.uid,
+      name: u.displayName,
+      avatar: u.photoURL,
+    };
+    console.log(newUser);
+    await Api.addUser(newUser);
+    //
+    setUser(newUser);
+  };
+
+  if (user === null) {
+    return "" + "";
+  }
+
+  return (
+    <AppWindow>
+      <Sidebar>
+        <NewChat
+          chatlist={chatlist}
+          user={user}
+          show={showNewChat}
+          setShow={setshowNewChat}
+        />
+        <SidebarHeader>
+          <Avatar src={user.avatar} alt="" />
```

Comment:
This user can be null, add this avatar conditionaly and it should work

Reply:
```<Avatar>
            {props.user && props.user.photoURL ? (
              <img src={props.user.photoURL} alt="" />
            ) : (
              <img src="/images/user.svg" alt="" />
            )}
          </Avatar>
```

How about this?

Reply:
great, I think that's all, is it working?

Reply:
I'm telling you what, this private messages in getting under my skin. I've made 15 private messaging apps that all work fine but there in firebase v9 and do not work with the website. I need to stop being a wuss and take the firebase course from educative.io. I've been working on this for what, 3 weeks now